### PR TITLE
CVE-2007-0404 and CVE-2009-2693 

### PR DESCRIPTION
**Pull request by ADAdebomi**
Figuring out how to find commits from the SVN revision

### CODE REVIEW

File: cves/CVE-2009-2693.yml [hunk 1]
```diff
@@ -43,51 +43,68 @@ description_instructions: |
...
-  command can be used to modify files outside of the web root directory. The 
-  files can be created, modified, or deleted. 
-  Secure information from the web server could be moved out of sccope, or a key
-  file for the server could be deleted which could cause a denial of service.
+description:
+  This vulnerability came up in Apache Tomcat 5 and 6 and allowed attackers to create,
+    delete, alter files, view paths and gain access to other parts of the system they should not have permission to
+    by creating special WAR (Web Application Resource) files to be deployed on the Tomcat Server. This occurred because
+    Tomcat did not check these files for directory traversal attempts. If successful the system would give
+    the attackers to any if not all directories accessible to the Tomcat process. Including "../../bin/catalina.bat" as an enty
```
Comment (dwelcheck) [inline cves/CVE-2009-2693.yml]: I think you are missing a word here
> "If successful the system would give the attackers (access) to any if not all directories...

File: cves/CVE-2009-2693.yml [hunk 2]
```diff
@@ -43,51 +43,68 @@ description_instructions: |
...
+
 
 upvotes_instructions: |
   Students: when initially writing this, ignore this upvotes number.
   Once this work is being reviewed, you will be giving a certain amount of
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 5
+upvotes:
```
Comment (dwelcheck) [inline cves/CVE-2009-2693.yml]: +5

File: cves/CVE-2009-2693.yml [hunk 3]
```diff
@@ -144,11 +160,14 @@ interesting_commits:
     emerging themes?
     If there are no interesting commits, demonstrate that you completed this
     section by explaining what happened between the VCCs and the fix.
-  answer:
+  answer: The methods required for validation was created between the VCC and the fix and also implemented
```
Comment (dwelcheck) [inline cves/CVE-2009-2693.yml]: Some wrong verb tensing here.  Correct it to: "The methods required for validation were created and implemented between the VCC and the fix".

File: cves/CVE-2009-2693.yml [hunk 4]
```diff
@@ -162,24 +181,23 @@ lessons:
...
   native_wrappers:
-    applies: false
+    applies:
     note:
   distrust_input:
-    applies: false
-    note:
+    applies: true
+    note: If the input were distrusted and the validate method was always in place, this vulnerability
+      woudl not have occurred
```
Comment (dwelcheck) [inline cves/CVE-2009-2693.yml]: Typo: **would** not have occurred

File: cves/CVE-2009-2693.yml [hunk 5]
```diff
@@ -209,10 +231,7 @@ mistakes:
...
-    that someone might try to navigate outside of the root directory and never
-    put in test cases and exceptions for when that happens. Since there was a lack
-    of unit testing, the directory traversal was never tested for and stayed in the 
-    code for a while unnoticed. This isn't exactly a mistake but more of an oversight 
-    on the developers part. The vulnerability was fixed by adding checks for "../" 
-    for any path that was being constructed. 
+  answer: It was a very simple mistake that was made. The developer did not take 
+    validating the file names before running them which was the error. The fix was
+    a quick one that implemented the mitigations that was suggested in the CWE related
+    to this vulnerability
```
Comment (dwelcheck) [inline cves/CVE-2009-2693.yml]: Some incorrect verb tensing and a missing period:

_The fix was a quick one that implemented the mitigations that **were** suggested in the CWE related to this vulnerability._

File: cves/CVE-2009-2693.yml [hunk 6]
```diff
@@ -209,10 +231,7 @@ mistakes:
...
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: The mistake that was made here was simple. The developer neglected to consider
-    that someone might try to navigate outside of the root directory and never
-    put in test cases and exceptions for when that happens. Since there was a lack
-    of unit testing, the directory traversal was never tested for and stayed in the 
-    code for a while unnoticed. This isn't exactly a mistake but more of an oversight 
-    on the developers part. The vulnerability was fixed by adding checks for "../" 
-    for any path that was being constructed. 
+  answer: It was a very simple mistake that was made. The developer did not take
```
Comment (dwelcheck) [inline cves/CVE-2009-2693.yml]: Try to make this sentence flow a bit better by getting rid of the passive voice.  Maybe:

_The mistake made was very simple._

File: cves/CVE-2009-2693.yml [hunk 7]
```diff
@@ -209,10 +231,7 @@ mistakes:
...
     engineering industry would find interesting.
-  answer: The mistake that was made here was simple. The developer neglected to consider
-    that someone might try to navigate outside of the root directory and never
-    put in test cases and exceptions for when that happens. Since there was a lack
-    of unit testing, the directory traversal was never tested for and stayed in the 
-    code for a while unnoticed. This isn't exactly a mistake but more of an oversight 
-    on the developers part. The vulnerability was fixed by adding checks for "../" 
-    for any path that was being constructed. 
+  answer: It was a very simple mistake that was made. The developer did not take 
+    validating the file names before running them which was the error. The fix was
```
Comment (dwelcheck) [inline cves/CVE-2009-2693.yml]: This phrasing is a bit weird:
>The developer did not take validating the file names

File: cves/CVE-2009-2693.yml (hunk 1 above)
Comment (ADAdebomi) [reply cves/CVE-2009-2693.yml]: Ah yes forgot to add the word access. Thanks

File: cves/CVE-2009-2693.yml (hunk 3 above)
Comment (ADAdebomi) [reply cves/CVE-2009-2693.yml]: It was only one method so it should have been method and was not method and were. Thanks

File: cves/CVE-2009-2693.yml (hunk 4 above)
Comment (ADAdebomi) [reply cves/CVE-2009-2693.yml]: Fixed it, thanks

File: cves/CVE-2009-2693.yml (hunk 6 above)
Comment (ADAdebomi) [reply cves/CVE-2009-2693.yml]: Fixed, thanks

File: cves/CVE-2009-2693.yml (hunk 7 above)
Comment (ADAdebomi) [reply cves/CVE-2009-2693.yml]: Correct, rephrased and fixed. Thanks

File: cves/CVE-2009-2693.yml (hunk 5 above)
Comment (ADAdebomi) [reply cves/CVE-2009-2693.yml]: Corrected wording, thanks

**Pull request merged**